manager/integration/tests/test_snapshot.py

```diff
@@ -0,0 +1,598 @@
+                assert checksum == s.checksum
+
+                # Check ctime in checksum file and from stat are matched
+                ctime_recorded = get_ctime_in_checksum_file(disk_path)
+                ctime = get_ctime_from_snapshot_disk_file(data_path, s.name)
+
+                print(f'snapshot {s.name}: '
+                      f'ctime recorded={ctime_recorded}, '
+                      f'ctime={ctime}')
+
+                df = pd.DataFrame({'string': [ctime_recorded, ctime]})
+                df['timestamp'] = pd.to_datetime(df['string'].str.replace(" +0000", "", regex=False))  # NOQA
+                assert str(df['timestamp'][0]) == str(df['timestamp'][1])
+
+            if snapshots.index(s) == len(snapshots)-1:
+                return True
+
+        time.sleep(RETRY_INTERVAL)
+
+    return False
+
+
+def get_checksum_in_checksum_file(disk_path):
+    checksum_file = disk_path + ".checksum"
+    try:
+        f = open(checksum_file)
+        metadata = json.load(f)
+    except OSError as e:
+        print(e)
+        raise
+
+    return metadata["checksum"]
+
+
+def get_ctime_in_checksum_file(disk_path):
+    checksum_file = disk_path + ".checksum"
+    try:
+        f = open(checksum_file)
+        metadata = json.load(f)
+    except OSError as e:
+        print(e)
+        raise
+
+    return metadata["change_time"]
+
+
+def check_snapshot_checksums_set(volume):
+    value = volume.snapshotList()
+    snapshots = value.data
+    for s in snapshots:
+        if s.name == "volume-head":
+            continue
+        if s.checksum == "":
+            return False
+
+    for s in snapshots:
+        print(f'Checksum of snapshot {s.name} is {s.checksum}')
+    return True
+
+
+def get_local_host_replica_data_path(volume):
+    lht_hostId = get_self_host_id()
+
+    dataPath = ""
+    replicas = volume.replicas
+    for r in replicas:
+        if r.hostId == lht_hostId:
+            dataPath = r.dataPath
+            break
+
+    return dataPath
+
+
+def get_checksum_from_snapshot_disk_file(data_path, snapshot_name):
+    exec_cmd = ["java", "-jar", "/jacksum.jar", "-a", "crc64_go-iso",
+                os.path.join(data_path,
+                             "volume-snap-" + snapshot_name + ".img")]
+    try:
+        output = subprocess.check_output(exec_cmd)
+    except subprocess.CalledProcessError as e:
+        print(e.output)
+        return ""
+
+    data = str(output, 'utf-8')
+    fields = data.split(' ')
+    return fields[0]
+
+
+def get_ctime_from_snapshot_disk_file(data_path, snapshot_name):
+    exec_cmd = ["stat", "-c", "%z",
+                os.path.join(data_path,
+                             "volume-snap-" + snapshot_name + ".img")]
+    try:
+        output = subprocess.check_output(exec_cmd)
+    except subprocess.CalledProcessError as e:
+        print(e.output)
+        return ""
+
+    return str(output, 'utf-8').rstrip('\n') + " UTC"
+
+
+def create_snapshots(client, volume, data_size):  # NOQA
```

Comment:
Can we make this function definition more generic? ex: `get_ctime_from_snapshot_disk_file(data_path, snapshot_name, num_snapshots)`

Reply:
Do you mean get multiple snapshots' ctimes at the same time?

Reply:
Apology for the confusion, that is not what I meant and I've updated my previous comment example.

Reply:
Updated. Thank you.

Reply:
Where is this updated?

Reply:
Sorry...forgot to push it...